api/flaskr/files.py

```diff
@@ -56,17 +57,32 @@ def upload_terrain_data() -> Response:
         return Response("No file(s) uploaded", status=400)
 
     valid_uploads = list(filter(is_valid_upload, uploaded_files))
-    if not valid_uploads:
+    if not valid_uploads or len(valid_uploads) != 2:
         print("No valid uploads")
         return Response('invalid image(s)', 400)
 
-    for upload in valid_uploads:
+    paths = []
+    pcdname = ""
+    for i, upload in enumerate(valid_uploads):
```

Comment:
Why do you loop but then just use one file?

Reply:
We expect there to be two files, we loop through both of them but if any of theme have an invalid name, we return